appinventor/appengine/src/com/google/appinventor/client/DesignToolbar.java

```diff
@@ -94,16 +94,19 @@ public DesignProject(String name, long projectId) {
 
     // Returns true if we added the screen (it didn't previously exist), false otherwise.
     public boolean addScreen(String name, FileEditor formEditor, FileEditor blocksEditor) {
-      if (!screens.containsKey(name)) {
-        screens.put(name, new Screen(name, formEditor, blocksEditor));
-        return true;
-      } else {
+      if (screens.containsKey(name)) {
         return false;
       }
+      screens.put(name, new Screen(name, formEditor, blocksEditor));
+      ((YaBlocksEditor) blocksEditor).addScreen(name);
+      return true;
     }
 
     public void removeScreen(String name) {
-      screens.remove(name);
+      if (screens.containsKey(name)) {
```

Comment:
If my advice above is taken, this can revert back as `YaBlocksEditor` will be informed via `onProjectNodeRemoved`.